client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -188,6 +216,7 @@ const PluginsBrowser = ( { trackPageViews = true, category, search } ) => {
 				search={ search }
 			/>
 			<div className="plugins-browser__content-wrapper">
+				{ ( category || search ) && renderBackLink() }
```

Comment:
Why we show the back button on search too? Is not enough to use the X icon to clear the search input?

<img width="311" alt="Screenshot 2024-08-29 at 12 44 28" src="https://github.com/user-attachments/assets/6ad9adb6-7a0a-4313-bb80-ab7b210d26f7">

Reply:
Also noted that the Back button shows up when I switch from "Discover" to other category from the top (e.g. Search Engine Optimization).

Due to the current position of the back button, the UI is a bit janky with sudden jumps:

https://github.com/user-attachments/assets/fdbcdd9a-1cd6-4ea7-8eae-7dd63ceb96e9


I think we should avoid displaying the Back button when activating one of these categories. We should display it only when using the "Browse all" links:

<img width="944" alt="Screenshot 2024-08-29 at 12 46 35" src="https://github.com/user-attachments/assets/ae06a1e0-2861-4e23-8355-0db55330007c">

Reply:
> Why we show the back button on search too? Is not enough to use the X icon to clear the search input?

Technically yes, but from a user experience point of view I wondered if that was a bit too subtle for lots of folk.

> I think we should avoid displaying the Back button when activating one of these categories. We should display it only when using the "Browse all" links:

Because the current implementation is janky or because you don't think there's a problem with users figuring out how to go back? 🤔 I think realising you have to click on "Discover" is a bit tricky and this categories confusion is what the [initial report](https://github.com/Automattic/wp-calypso/issues/93588) was all about.

Reply:
>  this categories confusion is what the [initial report](https://github.com/Automattic/wp-calypso/issues/93588) was all about.

I think the issue only reported a problem with the "Browse all" links:

> users can click on **"Browse All"** to see the full list. [...] once they enter the subsection, there is no back button

Reply:
To expand on the above, the report comes from this walkthrough: p9Jlb4-dr7-p2#comment-12794

> No “back” navigation when you click on a Browse all and go to a sub-screen at /browse/{term}.

Reply:
> i.e. the back button should only appear in sections that can be only accessed from a "Browse all" button: Must-have premium plugins, Supercharging and monetizing your blog, etc.

It's the same scenario.

Clicking "Browse all" for "Supercharging and monetizing your blog" goes to https://wordpress.com/plugins/browse/monetization.

Clicking the "Monetization" category goes to https://wordpress.com/plugins/browse/monetization.

Both links display the same thing.

Reply:
I think we have three options

 1. Accept the jankiness (the opposite of what a quality focus should be doing). Not really an option.
 2. Move Back to be underneath the search/category strip to avoid the jankiness
 3. Kick it back to design to figure out a different solution.

Reply:
<img width="1461" alt="Screenshot 2024-08-29 at 16 00 59" src="https://github.com/user-attachments/assets/b759793e-8823-4698-8206-496810b78cc5">

Definitely feels visually off to me

Reply:
Yeah, I agree with you, it feels visually pretty off to me too. My preferred option would be to "Kick it back to design to figure out a different solution", but ideally something that is a bit more simpler than the ideas that were discussed in https://github.com/Automattic/wp-calypso/issues/93588.

If that's not possible, all I'm saying is that maybe we have a 4th option:

- Accept the jankiness on the categories with a "browse all" link


> It's the same scenario.

Except for one detail: "Monetization" is a category that is displayed in the same horizontal list as "Discover", which is initially selected. If a user switches categories there, they are more likely to observe that the current category is "Discover" and they are _deactivating_ it when selecting another category. Therefore, they are more likely to know that to go back or reset the results, they have to click "Discover".

That's not the case when they use the "Browse all" link because they are in different place than the horizontal list of categories.

That's why I think we don't need a back button when activating categories from the horizontal list.